mantidproject/mantid
# MatchPeaks patch

This PR adds one more option to new `MatchPeaks` algorithm to roll the peaks in the first workspace with the difference of the peak positions between the second and third workspaces (which will be needed for IN16B reduction at ILL). It also does some code clean-up along the way.

**To test:**
1. Try the algorithm out with some workspaces with one peaks
2. Make sure the results are consistent with the documentation



Fixes #17742 .



---
#### Reviewer

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):
##### Code Review
##### Functional Tests
- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.
